### PR TITLE
[Merged by Bors] - chore: use mathport output in FieldTheory.SplittingField.Construction

### PR DESCRIPTION
**Pull request by riccardobrasca**
The mathlib3 version was manually backported from the mathlib4 version. We use here the mathport output.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/FieldTheory/SplittingField/Construction.lean [hunk 1]
```diff
@@ -24,7 +22,13 @@ In this file we prove the existence and uniqueness of splitting fields.
 ## Main statements
 
 * `Polynomial.IsSplittingField.algEquiv`: Every splitting field of a polynomial `f` is isomorphic
-  to `SplittingField f` and thus, being a splitting field is unique up to isomorphism.
+  to `splitting_field f` and thus, being a splitting field is unique up to isomorphism.
```
Comment (Ruben-VandeVelde) [inline Mathlib/FieldTheory/SplittingField/Construction.lean:25]: Revert

File: Mathlib/FieldTheory/SplittingField/Construction.lean [hunk 2]
```diff
@@ -99,16 +103,15 @@ theorem natDegree_removeFactor (f : K[X]) : f.removeFactor.natDegree = f.natDegr
 #align polynomial.nat_degree_remove_factor Polynomial.natDegree_removeFactor
 
 theorem natDegree_removeFactor' {f : K[X]} {n : ℕ} (hfn : f.natDegree = n + 1) :
-    f.removeFactor.natDegree = n := by rw [natDegree_removeFactor, hfn, n.add_sub_cancel]
+  f.removeFactor.natDegree = n := by rw [natDegree_removeFactor, hfn, n.add_sub_cancel]
```
Comment (Ruben-VandeVelde) [inline Mathlib/FieldTheory/SplittingField/Construction.lean:106]: Is this right?

File: Mathlib/FieldTheory/SplittingField/Construction.lean [hunk 3]
```diff
@@ -120,17 +123,25 @@ def SplittingFieldAuxAux (n : ℕ) : ∀ {K : Type u} [Field K], ∀ _ : K[X],
       let ⟨L, fL, _⟩ := ih f.removeFactor
       ⟨L, fL, (RingHom.comp (algebraMap _ _) (AdjoinRoot.of f.factor)).toAlgebra⟩
 
+/-- Auxiliary construction to a splitting field of a polynomial, which removes
+`n` (arbitrarily-chosen) factors. It is the type constructed in `splitting_field_aux_aux`.
```
Comment (Ruben-VandeVelde) [inline Mathlib/FieldTheory/SplittingField/Construction.lean:127]: SplittingFieldAuxAux

File: Mathlib/FieldTheory/SplittingField/Construction.lean [hunk 4]
```diff
@@ -157,7 +168,6 @@ section LiftInstances
 #noalign polynomial.splitting_field_aux.Field
 #noalign polynomial.splitting_field_aux.inhabited
 #noalign polynomial.splitting_field_aux.mk_hom
```
Comment (Ruben-VandeVelde) [inline Mathlib/FieldTheory/SplittingField/Construction.lean:170]: Should these be removed?

File: Mathlib/FieldTheory/SplittingField/Construction.lean [hunk 5]
```diff
@@ -236,13 +242,16 @@ theorem adjoin_rootSet (n : ℕ) :
     rw [ih _ (natDegree_removeFactor' hfn), Subalgebra.restrictScalars_top]
 #align polynomial.splitting_field_aux.adjoin_root_set Polynomial.SplittingFieldAux.adjoin_rootSet
 
-instance : IsSplittingField K (SplittingFieldAux f.natDegree f) f :=
+instance (f : K[X]) : IsSplittingField K (SplittingFieldAux f.natDegree f) f :=
   ⟨SplittingFieldAux.splits _ _ rfl, SplittingFieldAux.adjoin_rootSet _ _ rfl⟩
 
+/-- The natural map from `mv_polynomial (f.root_set (splitting_field_aux f.nat_degree f))`
+to `splitting_field_aux f.nat_degree f` sendind a variable to the corresponding root. -/
```
Comment (Ruben-VandeVelde) [inline Mathlib/FieldTheory/SplittingField/Construction.lean:249]: Fix names in these two lines

File: Mathlib/FieldTheory/SplittingField/Construction.lean [hunk 6]
```diff
@@ -252,17 +261,23 @@ theorem ofMvPolynomial_surjective (f : K[X]) : Function.Surjective (ofMvPolynomi
   intro α hα
   apply Algebra.subset_adjoin
   exact ⟨⟨α, hα⟩, rfl⟩
+#align polynomial.splitting_field_aux.of_mv_polynomial_surjective Polynomial.SplittingFieldAux.ofMvPolynomial_surjective
 
+/-- The algebra isomorphism between the quotient of
+`mv_polynomial (f.root_set (splitting_field_aux f.nat_degree f)) K` by the kernel of
+`of_mv_polynomial f` and `splitting_field_aux f.nat_degree f`. It is used to transport all the
+algebraic structures from the latter to `f.splitting_field`, that is defined as the former. -/
```
Comment (Ruben-VandeVelde) [inline Mathlib/FieldTheory/SplittingField/Construction.lean:269]: Here too

File: Mathlib/FieldTheory/SplittingField/Construction.lean [hunk 7]
```diff
@@ -324,7 +348,7 @@ instance [CharZero K] : CharZero (SplittingField f) :=
   charZero_of_injective_algebraMap (algebraMap K _).injective
 
 -- The algebra instance deriving from `K` should be definitionally equal to that
--- deriving from the field structure on `SplittingField f`.
+-- deriving from the field structure on `splitting_field f`.
```
Comment (Ruben-VandeVelde) [inline Mathlib/FieldTheory/SplittingField/Construction.lean:351]: Revert

File: Mathlib/FieldTheory/SplittingField/Construction.lean [hunk 8]
```diff
@@ -380,7 +399,13 @@ variable {K}
...
   finiteDimensional f.SplittingField f
 
-/-- Any splitting field is isomorphic to `SplittingField f`. -/
+instance [Fintype K] (f : K[X]) : Fintype f.SplittingField :=
+  FiniteDimensional.fintypeOfFintype K _
+
+instance (f : K[X]) : NoZeroSMulDivisors K f.SplittingField :=
+  inferInstance
+
+/-- Any splitting field is isomorphic to `splitting_field f`. -/
```
Comment (Ruben-VandeVelde) [inline Mathlib/FieldTheory/SplittingField/Construction.lean:408]: Revert

File: Mathlib/FieldTheory/SplittingField/Construction.lean (hunk 2 above)
Comment (riccardobrasca) [reply Mathlib/FieldTheory/SplittingField/Construction.lean:106]: You mean the indentation? Yes it looks correct (it was 4 characters before).

File: Mathlib/FieldTheory/SplittingField/Construction.lean (hunk 4 above)
Comment (riccardobrasca) [reply Mathlib/FieldTheory/SplittingField/Construction.lean:171]: OK I am stupid, of course they should be removed now.

**Review by jcommelin — APPROVED** [review]
LGTM

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**